R/chat.R

```diff
@@ -323,16 +325,35 @@ Chat <- R6::R6Class(
       ok <- !map_lgl(resps, is.null)
       json <- map(resps[ok], resp_body_json)
 
-      rows <- map(json, function(json) {
-        turn <- value_turn(private$provider, json, has_type = TRUE)
+      turns <- map(json, function(json) {
+        value_turn(private$provider, json, has_type = TRUE)
+      })
+      rows <- map(turns, function(turn) {
         extract_data(turn, type, convert = FALSE, needs_wrapper = needs_wrapper)
       })
 
       if (convert) {
-        convert_from_type(rows, type_array(items = type))
+        out <- convert_from_type(rows, type_array(items = type))
       } else {
-        rows
+        out <- rows
       }
+
+      if (is.data.frame(out) && (include_tokens || include_cost)) {
```

Comment:
Would it make sense to attach tokens and cost as an attribute to each item if not a data frame?

Or alternately maybe a warning if `out` isn't a data frame but tokens or cost were requested.

Reply:
We could do either but they feel a bit like overkill to me at the moment. I do believe that 95% of uses will yield a data frame so the carve out is mostly just to be safe.